iec61850serversimulator/src/main/java/com/cgi/iec61850serversimulator/functionclass/EventDataListener.java

```diff
@@ -173,9 +180,9 @@ public EventDataListener(final Device device, final Scheduler scheduler, final D
                         // Sends updated Schedule to Schedule object
                         this.device.getRelay(relayNr).getSchedule(scheduleNr).setEnabled(((BdaBoolean) bda).getValue());
 
-                        // Sends updated Schedule to database
-                        this.databaseUtils
-                                .updateDatabaseSchedule(this.device.getRelay(relayNr).getSchedule(scheduleNr));
+                        // Puts to update schedule in HashMap
+                        toUpdateSchedules.put(relaySchedulePairs,
+                                this.device.getRelay(relayNr).getSchedule(scheduleNr));
```

Comment:
If you add something like this in the beginning of the for-loop, you can simplify lots of "this.device.getRelay(relayNr).getSchedule(scheduleNr)" calls to just "scheduleToUpdate":
Schedule scheduleToUpdate;
"if (relayNr > 0 && scheduleNr > 0) {
  scheduleToUpdate = this.device.getRelay(relayNr).getSchedule(scheduleNr);
}"

Reply:
Done